### PR TITLE
Add locale in DatePickerThemeData

### LINKED ISSUE #148202: locale in DatePickerThemeData
### Use case

when using `showDatePicker` we have the option to set `locale`
```
showDatePicker(
  locale: Locale('en'),
  ...
```
But when we have multiple of this widget across our app setting this locale individually is cumbersome and feels against best practices.
We do have a class to set the global theme `DatePickerThemeData`, but this class does not support `locale`.

### Proposal

Include `locale` property in `DatePickerThemeData` so that we may globaly set the `locale` for `showDatePicker`.

### PR DESCRIPTION
**Pull request by MajdSallora**
*This PR changes the date picker were add locale in DatePickerThemeData*

*List which issues are fixed by this PR. You must list at least one issue. An issue is not required if the PR fixes something trivial like a typo.*
Fixes #148202 

*If you had to change anything in the [flutter/tests] repo, include a link to the migration guide as per the [breaking change policy].*

## Pre-launch Checklist

- [X] I read the [Contributor Guide] and followed the process outlined there for submitting PRs.
- [X] I read the [Tree Hygiene] wiki page, which explains my responsibilities.
- [X] I read and followed the [Flutter Style Guide], including [Features we expect every widget to implement].
- [X] I signed the [CLA].
- [X] I listed at least one issue that this PR fixes in the description above.
- [X] I updated/added relevant documentation (doc comments with `///`).
- [X] I added new tests to check the change I am making, or this PR is [test-exempt].
- [X] I followed the [breaking change policy] and added [Data Driven Fixes] where supported.
- [x] All existing and new tests are passing.

If you need help, consider asking for advice on the #hackers-new channel on [Discord].

[Contributor Guide]: https://github.com/flutter/flutter/wiki/Tree-hygiene#overview
[Tree Hygiene]: https://github.com/flutter/flutter/wiki/Tree-hygiene
[test-exempt]: https://github.com/flutter/flutter/wiki/Tree-hygiene#tests
[Flutter Style Guide]: https://github.com/flutter/flutter/wiki/Style-guide-for-Flutter-repo
[Features we expect every widget to implement]: https://github.com/flutter/flutter/wiki/Style-guide-for-Flutter-repo#features-we-expect-every-widget-to-implement
[CLA]: https://cla.developers.google.com/
[flutter/tests]: https://github.com/flutter/tests
[breaking change policy]: https://github.com/flutter/flutter/wiki/Tree-hygiene#handling-breaking-changes
[Discord]: https://github.com/flutter/flutter/wiki/Chat
[Data Driven Fixes]: https://github.com/flutter/flutter/wiki/Data-driven-Fixes

### CODE REVIEW

File: packages/flutter/lib/src/material/date_picker.dart [hunk 1]
```diff
@@ -241,6 +241,15 @@ Future<DateTime?> showDatePicker({
       locale: locale,
       child: dialog,
     );
+  }else{
```
Comment (MitchellGoodwin) [inline packages/flutter/lib/src/material/date_picker.dart:244]:
```suggestion
  } else {
```
nit: spacing

**Review by MitchellGoodwin — COMMENTED** [review]
Mostly LGTM with one nit. Thank you for putting together this fix.

File: packages/flutter/lib/src/material/date_picker.dart (hunk 1 above)
Comment (MajdSallora) [reply packages/flutter/lib/src/material/date_picker.dart]: Thanks for your suggestion, I've commit it
@MitchellGoodwin

**Review by MitchellGoodwin — APPROVED** [review]
LGTM! Thank you again for the PR. This needs a second approval, so let me see if I can find someone.

**Review by justinmc — APPROVED** [review]
LGTM :+1:

**Pull request merged**